storage/innobase/buf/buf0buf.cc

```diff
@@ -4893,6 +4897,13 @@ bool buf_page_io_complete(buf_page_t *bpage, bool evict) {
       ut_error;
   }
 
+  if (io_type == BUF_IO_WRITE && fsp_is_system_temporary(bpage->id.space())) {
+    const auto it = innodb_temp_table_thds.find(bpage->id.space());
```

Comment:
I think you need the lock to call find() since it's not thread-safe? But it would also lead to every temp table write synchronizing on this mutex. Maybe shard the lock based on space_id? Or if there's another way to track writes on the space_id and then update the session variable at a later point without having to acquire a lock here?

Reply:
You are right, I missed this. I added this map to avoid a global lock (there is already an existing list which requires that in the codebase), but I think I can fix it to work without it. Will update the PR later today.